src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryPublisher.cs

```diff
@@ -41,9 +41,9 @@ public ValueTask PublishEvent(EvaluationEvent evaluationEvent, CancellationToken
                 properties["Variant"] = evaluationEvent.Variant.Name;
             }
 
-            if (featureDefinition.TelemetryMetadata != null)
+            if (featureDefinition.Telemetry.Metadata != null)
```

Comment:
Is it necessary to add null check for `featureDefinition.Telemetry`?

Reply:
This method should not be invoked if telemetry is null. I will update to throw an argument exception if telemetry is null.

Reply:
Updated.